test/utils/deploy/openshift.go

```
@@ -0,0 +1,130 @@
+// other dependencies, so the controller will be able to install TAS
+// stack properly
+func (o *OpenShiftNRO) Deploy() {
+	GinkgoHelper()
+
+	// TODO: should this be configurable?
+	o.deployWithLabels(objects.OpenshiftMatchLabels())
+}
+
+func (o *OpenShiftNRO) deployWithLabels(matchLabels map[string]string) {
+	GinkgoHelper()
+	nroObj := objects.TestNRO(matchLabels)
+	kcObj, err := objects.TestKC(matchLabels)
+	Expect(err).To(Not(HaveOccurred()))
+
+	unpause, err := e2epause.MachineConfigPoolsByNodeGroups(nroObj.Spec.NodeGroups)
+	Expect(err).NotTo(HaveOccurred())
+
+	var createKubelet bool
+	if _, ok := os.LookupEnv("E2E_NROP_INSTALL_SKIP_KC"); ok {
+		By("using cluster kubeletconfig (if any)")
+	} else {
+		By(fmt.Sprintf("creating the KC object: %s", kcObj.Name))
+		err = e2eclient.Client.Create(context.TODO(), kcObj)
+		Expect(err).NotTo(HaveOccurred())
+		o.KcObj = kcObj
+		createKubelet = true
+	}
+
+	By(fmt.Sprintf("creating the NRO object: %s", nroObj.Name))
+	err = e2eclient.Client.Create(context.TODO(), nroObj)
+	Expect(err).NotTo(HaveOccurred())
+	o.NroObj = nroObj
+
+	Eventually(unpause).WithTimeout(configuration.MachineConfigPoolUpdateTimeout).WithPolling(configuration.MachineConfigPoolUpdateInterval).ShouldNot(HaveOccurred())
+
+	err = e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroObj), nroObj)
+	Expect(err).NotTo(HaveOccurred())
+	o.NroObj = nroObj
+
+	if createKubelet || annotations.IsCustomPolicyEnabled(nroObj.Annotations) {
+		By("waiting for MCP to get updated")
+		mcps, err := nropmcp.GetListByNodeGroupsV1(context.TODO(), e2eclient.Client, nroObj.Spec.NodeGroups)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(WaitForMCPsCondition(e2eclient.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdating)).To(Succeed())
+		Expect(WaitForMCPsCondition(e2eclient.Client, context.TODO(), mcps, machineconfigv1.MachineConfigPoolUpdated)).To(Succeed())
+	}
+}
+
+// TODO: what if timeout < period?
```

fail loudly, 99/100 it's a CI config mistake, the sooner we notice we better